util/circular_buffer.c

```diff
@@ -146,12 +141,148 @@ int32_t cb_remove(struct circular_buffer *desc)
  *  - \ref SUCCESS : On success
  *  - \ref FAILURE : Otherwise
  */
-int32_t cb_size(struct circular_buffer *desc, uint32_t *nb_elements)
+int32_t cb_size(struct circular_buffer *desc, uint32_t *size)
+{
+	uint32_t overflows;
+	uint32_t overflowed_size;
+
+	if (!desc || !size)
+		return -EINVAL;
+
+	if (desc->write.nb_overflows < desc->read.nb_overflows)
+		overflows = UINT32_MAX - desc->read.nb_overflows +
+			    desc->write.nb_overflows;
```

Comment:
Would there be actually:
```suggestion
		overflows = UINT32_MAX - desc->read.nb_overflows +
			    desc->write.nb_overflows + 1;
```
To get the accurate number of overflows? Since `nb_overflows` starts at 0 after overflowing.

Reply:
nice catch

Reply:
solved